### PR TITLE
chore(popx): code simplification and refactoring

### PR DESCRIPTION
**Pull request by zepatrik**
While debugging the migrator, I stumbled upon a couple of improvements that could be made.
I also removed the pkgerx package, as we don't use that now for years.

### CODE REVIEW

File: popx/migrator.go [hunk 1]
```diff
@@ -101,44 +101,39 @@ func (m *Migrator) UpTo(ctx context.Context, step int) (applied int, err error)
 		for _, mi := range mfs {
 			l := m.l.WithField("version", mi.Version).WithField("migration_name", mi.Name).WithField("migration_file", mi.Path)
 
-			exists, err := c.Where("version = ?", mi.Version).Exists(mtn)
+			appliedMigrations := make([]string, 0, 2)
+			legacyVersion := mi.Version
+			if len(legacyVersion) > 14 {
+				legacyVersion = legacyVersion[:14]
+			}
+			err := c.RawQuery(fmt.Sprintf("SELECT version FROM %s WHERE version IN (?, ?)", mtn), mi.Version, legacyVersion).All(&appliedMigrations)
```
Comment (zepatrik) [inline popx/migrator.go:109]: We've now combined two queries into one, which should half the number of queries and (depending on the database) can significantly improve the performance of the migrate job.

File: popx/migrator.go [hunk 2]
```diff
@@ -596,13 +586,6 @@ func (m *Migrator) exec(ctx context.Context, fn func() error) error {
 		}
 	}
 
-	if m.Connection.Dialect.Name() == "cockroach" {
-		outer := fn
-		fn = func() error {
-			return crdb.Execute(outer)
```
Comment (zepatrik) [inline popx/migrator.go:602]: From the documentation of `crdb.Execute`:

> It is used to add retry handling to the execution of a single statement. If a multi-statement transaction is being run, use ExecuteTx instead.

But as the transaction is started within, we don't need to retry anything at this point.

File: popx/migrator_test.go [hunk 3]
```diff
@@ -4,144 +4,28 @@
...
 	"github.com/ory/x/logrusx"
-	"github.com/ory/x/pkgerx"
 	. "github.com/ory/x/popx"
-	"github.com/ory/x/sqlcon/dockertest"
 )
 
 //go:embed stub/migrations/transactional/*.sql
 var transactionalMigrations embed.FS
 
-func TestMigratorUpgrading(t *testing.T) {
```
Comment (zepatrik) [inline popx/migrator_test.go:29]: This test was mostly skipped anyway, and I removed the "legacy" migrator with this PR, so no need to test compatibility anymore.

**Review by aeneasr — APPROVED** [review]

File: popx/migrator.go (hunk 2 above)
Comment (aeneasr) [reply popx/migrator.go:602]: With my recent auto-commit changes, we should add retry to migrations that run without transactions (aka go migrations and auto commit transactions).

**Pull request merged**